src/FsAutoComplete/CodeFixes/AddTypeAliasToSignatureFile.fs

```diff
@@ -0,0 +1,95 @@
+module FsAutoComplete.CodeFix.AddTypeAliasToSignatureFile
+
+open FSharp.Compiler.Symbols
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text
+open FSharp.Compiler.CodeAnalysis
+open FsToolkit.ErrorHandling
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete.CodeFix.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+
+// TODO: add proper title for code fix
+let title = "AddTypeAliasToSignatureFile Codefix"
+
+let fix (getParseResultsForFile : GetParseResultsForFile) : CodeFix =
+    fun (codeActionParams : CodeActionParams) ->
+        asyncResult {
+            // TODO: check if the file even has a signature file
```

Comment:
@baronfel any pointers on creating a good helper for this?
This code fix should really only kick in if I'm targeting an implementation file that has a signature.

Reply:
The simplest way would be to do something like
* get source file path in question
* get LoadedProject for that file
* check if signature for that file exists in LoadedProject

If you have an `FSharpSymbol` I think there's a property that has the 'Declaration Location' that'll be an FSI if it exists, but that's way more expensive to compute so I think going with the above strategy is fine.

I wouldn't necessarily make it a 'helper' yet as this is the only CodeFix that wants this triggering condition, but once there's a second we could look at naming/extracting the pattern.

Reply:
Yeah, I need to access that property later on anyway to check if it is in a signature or not.
But would like to avoid that check if there is no signature at all.